### PR TITLE
Forward the websocket close event to application callbacks

### PR DESCRIPTION
**Pull request by yasushi-saito**
The close event contains contents of the close frame sent by the server. The close frame is used by the server to inform of the connection setup problem.

### CODE REVIEW

File: js/src/SmartConnect/api.md [hunk 1]
```diff
@@ -6,32 +6,35 @@ a sessionURL is already provided in the configuration
...
+## onConnectionClose(callback, err) : subscription
 
-Register callback for when the connection close.
+Register callback for when the connection close.  Err is a websocket event. If
+the server closes the connection after sending a close frame
+(https://datatracker.ietf.org/doc/html/rfc6455#section-5.5.1), the err object
+will have shape {code: number, reason: string}.
 
-## onConnectionError(callback) : subscription
+## onConnectionError(callback, err) : subscription
```
Comment (jourdain) [inline js/src/SmartConnect/api.md:30]: err is passed to the callback. It is not part of the profile of the `onConnectionError`

File: js/src/SmartConnect/api.md [hunk 2]
```diff
@@ -6,32 +6,35 @@ a sessionURL is already provided in the configuration
...
+## connect()
 
 Trigger the connection request.
 
 ## onConnectionReady(callback) : subscription
 
 Register callback for when the connection became ready.
 
-## onConnectionClose(callback) : subscription
+## onConnectionClose(callback, err) : subscription
```
Comment (jourdain) [inline js/src/SmartConnect/api.md:23]: err is passed to the callback. It is not part of the profile

File: js/src/SmartConnect/api.md (hunk 1 above)
Comment (yasushi-saito) [reply js/src/SmartConnect/api.md]: You are right. Rephrased the doc.

File: js/src/SmartConnect/api.md (hunk 2 above)
Comment (yasushi-saito) [reply js/src/SmartConnect/api.md]: Done.

**Comment by jourdain** [comment]
If you want a new JS version, you need to update the version of `js/package.json` manually. That repo is not fully automated between the JS and Python.

**Comment by yasushi-saito** [comment]
Ok. I updated the version to 0.1.16.

**Comment by jourdain** [comment]
I'll have to publish it now... I'll try to do it in the next hour.

**Comment by jourdain** [comment]
published...

**Pull request merged**